### DISCUSSION TITLE
Дальнейшее развитие проекта (флудильня)

### DISCUSSION THREAD

**Discussion by zetcamp** [Anonym-tsk/nfqws-keenetic#108] [Обсуждения]
Во-первых строках выражаю неимоверно громадную благодарность автору за "очеловечивание болванки" из проекта zapret (ни разу не умаляю работу самого bol-van - если б не его труд, всего этого не было бы вообще). Сильно упрощает первоначальную работу с пакетом такая автоматизация установки через opkg там, где это возможно.
Поторогал веб-морду ещё в первых реинкарнациях и вот самую свежую сегодня, где уже есть и механика обновления пакета - огонь!

В связи с этим решил создать таку тему-флудильню, где можно было бы предложить какие-то свои идеи. Может кто что путное предложит.

Начну с себя:
1) возможно ли добавить в веб-морде какой-то раздел "настройки", где собрать всё, что касается пакета. Сейчас это просто выпадающий список с командами типа  перезапуска/обновления и т.д.
А я про отдельную вкладку или ещё какой элемент web-ui, где собрать всё, что свящано с пакетом отдельными элементами. Например, кнопку "обновить пакет" с выхлопом opkg, как сейчас, какую-то галку "проверка обновлений", которая бы мониторила переодически релизы на гитхабе и выводила по нажатию кнопки "история обновлений" эту самую историю релизов, галку "автообновление" - которая бы сама обновляла пакет до свежей версии, если таковой выпал на гитхабе. Просто на отдельной вкладке или разделе модального окна больше места, чтоб уместить всё, что потом может народиться.
2) возможно ли добавить веб-морду для openwrt либо отдельным, как для кинетика сейчас, web-ui на том же 88-м порту, либо как модуль luci-app?

Я понимаю, что это работёнка та ещё - такие свистоперделки рисовать. Но предложение родило спрос, вот и задался вопросами.

**Comment by unameras** [comment]
За флудильню спасибо!
Автору тоже огромное спасибо (от товарищей, которые воспользовались - тоже), но я ему донаты кидаю в благодарность!
По поводу доработок: как бы не стало сильно "тяжелым" приложение с доработками и "свистоперделками". Но стоит признать - для многих в веб-морде конечно проще управлять хотя бы конфигом и списками. Тут тоже автору огромное уважение - все новые версии из коробки работают без всякого шаманства - обновил и работает.
Конечно, есть одна фишка, о которой прям мечтал в начале проекта: установка в один клик, например, запустил bat-ник или exe-шник, а он сам всё поставил)

**Reply by zetcamp** [reply]
> как бы не стало сильно "тяжелым" приложение с доработками и "свистоперделками"

Ну, сам пакет же не претерпит изменений, предложенные мной перделки - они для вебморды и тоже легковесны - так как ничего особо нового не предлагается, только немного другой логики для web-ui.

> запустил bat-ник или exe-шник, а он сам всё поставил

В теории - это реализуемо, во всяком случае, для опенврт - нужно немного помёта и веток, чтоб подключиться к узлу и выполнить на нём последовательность команд в авто режиме с минимальным участием пользователя. В 10-ке, как и в лине - SSH уже есть в терминале, осталось дело за малым.

**Comment by deadcxap** [comment]
Очень был бы рад веб-интерфейсу для asus-merlin прошивок, там есть функционал встраивания вкладок стороннего софта прямо в веб-интерфейс самого роутера (например как в https://github.com/de-vnull/vnstat-on-merlin)
Или хотя бы на его микросервере: https://github.com/RMerl/asuswrt-merlin.ng/wiki/Lighttpd-web-server-with-PHP-support-through-Entware

**Comment by unixty** [comment]
Возможно ли реализовать механизм обновления файлов zapret отдельно от всего репозитория? Так как сейчас нужно релиз дополнительный выкатывать, а zapret достаточно часто обновляется

**Reply by Anonym-tsk** [reply]
zapret обновляется редко. Обновления nfqws-keenetic выходят сразу.

**Comment by zetcamp** [comment]
> 2\. возможно ли добавить веб-морду для openwrt

Вопрос снят с релизом 2.5.11. Всё прекрасно работает (на первый взгляд). Благодарю.

**Comment by 777Evgeny777** [comment]
а для PHP7 нельзя заточить WEB, а то на опенврт 19 не ставиться ?

**Reply by Anonym-tsk** [reply]
Попробуйте так
```
opkg install php7-cgi php7-mod-session
opkg install nfqws-keenetic-web --force-depends
```

**Reply by 777Evgeny777** [reply]
выдает ошибки после устаноки web
Configuring nfqws-keenetic-web.
//usr/lib/opkg/info/nfqws-keenetic-web.postinst: line 4: service: not found
//usr/lib/opkg/info/nfqws-keenetic-web.postinst: line 5: service: not found
NFQWS Web Interface installed: http://192.168.88.1:90
Collected errors:
 * satisfy_dependencies_for: Cannot satisfy the following dependencies for nfqws-keenetic-web:
 *      php8-cgi
 *      php8-mod-session

в сервисах  lighttpd есть, команды(service lighttpd start , service lighttpd reload) с консоли проходят штатно, вэб страничка открывается ,но сождержимого файлов нет, действия сделать никакие не дает, кнопка управления не активна, версия  внизу справа отображается корректно.

root@OpenWrt:/usr/lib/opkg/info# cat ./nfqws-keenetic-web.postinst
#!/bin/sh
[ "${IPKG_NO_SCRIPT}" = "1" ] && exit 0
service lighttpd start
service lighttpd reload
ip_addr=$(ip -f inet addr show dev br-lan 2>/dev/null | grep inet | sed -n 's/.*inet \([0-9.]\+\).*/\1/p')
echo "NFQWS Web Interface installed: http://$ip_addr:90"
[ -f ${IPKG_INSTROOT}/lib/functions.sh ] || exit 0
. ${IPKG_INSTROOT}/lib/functions.sh
default_postinst $0 $@

root@OpenWrt:/usr/lib/opkg/info# ./nfqws-keenetic-web.postinst
./nfqws-keenetic-web.postinst: line 3: service: not found
./nfqws-keenetic-web.postinst: line 4: service: not found
NFQWS Web Interface installed: http://192.168.88.1:90

root@OpenWrt:/usr/lib/opkg/info# service
service "" not found, the following services are available:
boot            led             rssileds        uhttpd
cron            lighttpd        stubby          umount
dnsmasq         log             sysctl          urandom_seed
done            network         sysfixtime      urngd
dropbear        nfqws-keenetic  sysntpd
firewall        odhcpd          system
gpio_switch     rpcd            ucitrack

root@OpenWrt:/usr/lib/opkg/info# service lighttpd reload service lighttpd start
root@OpenWrt:/usr/lib/opkg/info#

**Reply by bear-gha** [reply]
Это проблема OpenWRT:

https://forum.openwrt.org/t/service-not-found-error/14384/13
https://github.com/acmesh-official/acme.sh/issues/4031
https://forum.openwrt.org/t/unable-to-use-service-inside-script-on-openwrt-18-06/174861

Решение:

> This problem only occurs in opwnrt.
> Solution is to replace
> --reloadcmd "service nginx force-reload"
> with
> --reloadcmd "/etc/init.d/nginx reload "

и еще вопрос, а конфиги тут лежат?

```
NFQWS_BIN="/usr/bin/nfqws"
CONFFILE="/etc/nfqws/nfqws.conf"
PIDFILE="/var/run/nfqws.pid"
```

**Reply by 777Evgeny777** [reply]
да пути к файлам верные, по поводу исапрвления проблемы не совсем Вас понял , что сделать ?
по внешним признакам складывается ощущение что не проходит процедура идентификации, т.к. окно с именем пользователя/паролем не выходит, сразу открывается пустая страница, кнопка сервисов не активна, номер версии корректен

**Reply by bear-gha** [reply]
Логи пхп смотрите.

**Comment by EasyKill33** [comment]
Может есть возможность реализовать что то типа hosts? Чтобы ещё и заблоченные по ip сайты открывались, добавив незаблокированные ip в hosts?

**Reply by Anonym-tsk** [reply]
На кинетике вы можете использовать команду `ip host`

**Reply by EasyKill33** [reply]
> На кинетике вы можете использовать команду `ip host`

Сейчас так и делаю. Хотелось бы просто вынести это в интерфейс для удобства, что то типа юзер листа, либо прямо в юзер листе сделать что если после домена написан ип то пускать этот домен через этот ип на всех устройствах.

**Reply by Anonym-tsk** [reply]
nfqws ничего не знает про dns и не может им управлять

**Reply by EasyKill33** [reply]
Понял принял, спасибо

**Comment by LordArrin** [comment]
Было бы здорово добавить https к веб интерфейсу. По долгу наиболее "шарящего" приходится настраивать родственникам все обходы, так было бы проще перенаправлять порт на свой сервак. Ставить прокси на роутер слишком тяжело выйдет.

Сертификаты можно было бы просто кинуть в папку на роутере и указать путь

**Reply by Anonym-tsk** [reply]
Под капотом lighttpd, можете сконфигурировать как нравится

**Comment by rec99** [comment]
Напишу вопрос тут, чтобы не засорять дискуссии.

Поставил nfqws-keenetic, все завелось отлично, но есть сервисы типа инстаграмма, которые хотелось бы заворачивать отдельно в впн (wg, поднят на кинетике).

Нашел такой [гайд с dnsmasq-full](https://github.com/Sharm/keenetic-domain-routing/blob/master/VPN.mdl)

nfqws-keenetic не должен конфликтовать с такой настройкой? В сетях только начинаю разбираться, но вижу что там тоже iptables перенастраивается.

**Reply by YanusBeda** [reply]
> Напишу вопрос тут, чтобы не засорять дискуссии.
> 
> Поставил nfqws-keenetic, все завелось отлично, но есть сервисы типа инстаграмма, которые хотелось бы заворачивать отдельно в впн (wg, поднят на кинетике).
> 
> Нашел такой [гайд с dnsmasq-full](https://github.com/Sharm/keenetic-domain-routing/blob/master/VPN.mdl)
> 
> nfqws-keenetic не должен конфликтовать с такой настройкой? В сетях только начинаю разбираться, но вижу что там тоже iptables перенастраивается.

На OpenWrt NFQWS можно сочетать с  [Podkop](https://github.com/itdoginfo/podkop) проверил на своем роутере, работает отлично, по крайней в режиме Proxy (Vless, Shadowsocks) точно работает.
Буду пробовать на неделе , на Кинетике с [КВАС](https://github.com/qzeleza/kvas) завести.

**Reply by rec99** [reply]
Я себе настроил через [ipset-dns](https://forum.keenetic.ru/topic/16257-ipset-dns-для-выборочного-роутинга/). Если не много доменов куда заворачивать трафик, то в целом хорошее решение. Если нужно будет масштабировать, тоже КВАС себе наметил на будущее.

**Comment by nialone** [comment]
Прикольная штука. СТавится просто. ЮТ работает из под коробки. ВОпрос: В пакете используются старая версия модуля или свой скрипт? В последних версиях запрета обновили логику работы и конфиг. Там уже нет NFQWS_ARGS, NFQWS_ARGS_QUIC, NFQWS_ARGS_UDP, используется NFQWS_OPT. 

Обновление планируется?

**Comment by JasderoFox** [comment]
Спасибо за флудильню. И раз такое дело, сразу же задам тут вопрос. Возможно ли как-то nfqws отключить на одном из устройств, подключенных к роутеру? Хочется на одном из компьютеров его отключить, ибо там nfqws больше мешается, чем помогает. Но при этом что бы на всех остальных устройствах, nfqws по прежнему работал.

**Reply by unameras** [reply]
Описано в инструкции:
"Можно сделать политику исключения, добавив в конфиг POLICY_EXCLUDE=1. Тогда будет обрабатываться трафик для всех устройств, кроме тех, что добавлены в политику NFQWS.
Имя политики можно изменить в конфиге, параметр POLICY_NAME."

**Reply by unameras** [reply]
Как политику на роутере настроить написано тут - https://support.keenetic.ru/giga/kn-1011/ru/17892-connection-policies.html

**Reply by JasderoFox** [reply]
> Описано в инструкции: "Можно сделать политику исключения, добавив в конфиг POLICY_EXCLUDE=1. Тогда будет обрабатываться трафик для всех устройств, кроме тех, что добавлены в политику NFQWS. Имя политики можно изменить в конфиге, параметр POLICY_NAME."

Ооо, это именно то, что нужно. Большое человеческое спасибо!

**Comment by H4pH** [comment]
Наткнулся на подобный проект https://github.com/IndeecFOX/zapret4rocket 
Правда что-то со стратегиями там сильно накручено, но может что-то интересное есть для развития текущего проекта :)
